Sources/NIOSSH/Connection State Machine/SSHConnectionStateMachine.swift

```diff
@@ -335,6 +335,7 @@ struct SSHConnectionStateMachine {
 
         case .active(var state):
             guard let message = try state.parser.nextPacket() else {
+                self.state = .active(state)
```

Comment:
this is not the only place where we do not update the state, but it's the only one that breaks transport security schemes that use `decodeFirstBlock`. I also find it quite tricky to write a test for this. It should not affect initial key exchange since it's not encrypted, but it might affect re-keying. I will be trying to test that hypothesis, but wanted to start a conversation earlier in case you might come up with something as well...

Reply:
Given that we know that `nextPacket` may mutate the state, we should probably unconditionally re-store the state in all locations, even when we can't easily write tests for this.

Reply:
For re-keying I think you could try cribbing on the existing tests for re-keying.

Reply:
done